src/components/SeldonImage/SeldonImage.tsx

```diff
@@ -91,16 +125,44 @@ const SeldonImage = memo(
       const { className: baseClassName, ...commonProps } = getCommonProps(props, 'SeldonImage');
       const imgRef = useRef<HTMLImageElement>(null);
 
-      const [loadingState, setLoadingState] = useState<'loading' | 'loaded' | 'error'>('loading');
+      const [loadingState, setLoadingState] = useState<'loading' | 'loaded' | 'error'>(() => {
+        if (isServer) {
+          return 'loading';
+        }
+        const element = document.getElementById(src);
+
+        if (element instanceof HTMLImageElement && !element.classList.contains(`${baseClassName}-img--hidden`)) {
+          return 'loaded';
+        }
+
+        const img = document.createElement('img');
```

Comment:
I'm a bit surprised you had to do this all over again and couldn't just do this in the `isImageValid`?

Reply:
tbh this part was just adding extra redundancy, probably could get away with removing it